bokehjs/src/coffee/models/ranges/data_range1d.ts

```diff
@@ -180,14 +206,19 @@ export class DataRange1d extends DataRange {
     return [start, end]
   }
 
-  update(bounds: Bounds, dimension: Dim, bounds_id: string): void {
+  update(bounds: Bounds, dimension: Dim, bounds_id: string, r: number): void {
     if (this.have_updated_interactively)
       return
 
     const renderers = this.computed_renderers()
 
     // update the raw data bounds for all renderers we care about
-    this._plot_bounds[bounds_id] = this._compute_plot_bounds(renderers, bounds)
+    let total_bounds = this._compute_plot_bounds(renderers, bounds)
+
+    if (r != null)
```

Comment:
This should have resulted in an error, because `r` is a `number`, not `number | null` (or similar), but apparently it compiles. Looks like another bug in the compiler. Anyway, given `r` optional nature, the signature of this method should be changed to `r?: number`. I would also change `r` to something more meaningful (`ratio`, `aspect_ratio`).